eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/inmemory/InMemoryEventStorageEngine.java

```diff
@@ -50,8 +50,19 @@
  */
 public class InMemoryEventStorageEngine implements EventStorageEngine {
 
-    private final NavigableMap<TrackingToken, TrackedEventMessage<?>> events = new ConcurrentSkipListMap<>();
-    private final Map<String, List<DomainEventMessage<?>>> snapshots = new ConcurrentHashMap<>();
+  protected final NavigableMap<TrackingToken, TrackedEventMessage<?>> events;
+  protected final Map<String, List<DomainEventMessage<?>>> snapshots;
+
+  public InMemoryEventStorageEngine() {
```

Comment:
For the majority of our infrastructure components we have the Builder pattern in place. If we stay with this format of providing the `events` and `snapshots` (which @m1l4n54v1c challenged in the other comment), then I'd very much like to follow this pattern too.

Reply:
Event with the builder pattern two issues remain: 1) keeping no-arg constructor for backward compatibility, 2) the one setting up the builder has control over the maps implmentation.

Reply:
1) As it's part of 4.1 as the PR points to `master`, I'd view this as not to big a deal. Especially as I think the assumption is fair __nobody__ uses this in production (I hope).
2) I do not fully understand how this is different from using the (no-op)constructor. Would you mind elaborating?

Reply:
One added reason for the Builder-pattern here, is that we can more easily expand the inner workings of the `InMemoryEventStorageEngine` in the future without ever breaking backwards compatibility.